### PR TITLE
[OSDEV-1006] Format response output

### PR DESCRIPTION
**Pull request by VadimKovalenkoSNF**
Follow-up PR for [OSDEV-1006](https://opensupplyhub.atlassian.net/browse/OSDEV-1006)
Restructure OpenSearch response to this format:
```
{
    "count": <Number (total hits in DB)>
    "data": <array[Object]>
}
```
The `<array[Object]>` is represented by array of production locations entries.

[OSDEV-1006]: https://opensupplyhub.atlassian.net/browse/OSDEV-1006?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/django/api/services/search.py [hunk 1]
```diff
@@ -6,6 +6,29 @@
 logger = logging.getLogger(__name__)
 
 
+def prepare_opensearch_response(response):
```
[the remark this answers is not in the document]
Comment (VadimKovalenkoSNF) [reply src/django/api/services/search.py]: My concern was that I won't be able to test this, but after some research I realized that I can re-write unit tests to check `prepare_opensearch_response` as private method of the `OpenSearchService`. Thank you for pointing on this, pls re-re-review.

**Review by protsack-stephan — DISMISSED** [review]
Good work!

**Comment by VadimKovalenkoSNF** [comment]
Fixed error with wrong OpenSearch client instantiation on the test runtime.

**Review by vladsha-dev — APPROVED** [review]
LGTM

**Review by mazursasha1990 — APPROVED** [review]
LGTM

**Pull request merged**